libs/cardano-ledger-binary/testlib/Test/Cardano/Ledger/Binary/RoundTrip.hs

```diff
@@ -84,6 +85,13 @@ roundTripFailureExpectation version x =
       expectationFailure $
         "Should not have deserialized: " ++ showExpr (CBORBytes (serialize' version x))
 
+-- Tests that a decoder error does not happen for a specific version
+roundTripVersionExpectation :: forall a. (Eq a, Show a, ToCBOR a, FromCBOR a) => Version -> a -> Expectation
+roundTripVersionExpectation version x =
+  case roundTrip version (cborTrip @a) x of
+    Left err -> expectationFailure $ "Failed to deserialize encoded: " ++ show err
+    Right tDecoded -> tDecoded `shouldBe` x
+
```

Comment:
I recommend you do this instead:

```
roundTripExpectation = roundTripRangeExpectation minBound maxBound

-- | Just like `roundTripExpectation`, but allows to supply an inclusive version range,
-- instead of testing all versions
roundTripRangeExpectation ::
  (Show t, Eq t, Typeable t, HasCallStack) =>
  -- | From version
  Version ->
  -- | To version
  Version ->
  Trip t t ->
  t ->
  Expectation
roundTripRangeExpectation fromVersion toVersion trip t =
  forM_ [fromVersion .. toVersion] $ \version ->
     case roundTrip version trip t of
       Left err -> expectationFailure $ "Failed to deserialize encoded: " ++ show err
       Right tDecoded -> tDecoded `shouldBe` t
```

This way whenever you want to test roundtrip with custom version range you can

Reply:
Awesome!